public/app/plugins/datasource/prometheus/language_provider.ts

```diff
@@ -109,6 +110,10 @@ export default class PromQlLanguageProvider extends LanguageProvider {
     this.timeRange = { start: 0, end: 0 };
     this.metrics = [];
 
+    this.keysCache = new LRU<string, string[]>({ max: this.datasource.getCacheLength() });
+    this.labelsCache = new LRU<string, Record<string, string[]>>({ max: this.datasource.getCacheLength() });
+    this.labelValuesCache = new LRU<string, string[]>({ max: this.datasource.getCacheLength() });
```

Comment:
There is also a maxSize which limits the total "length" of all of the values stored, but this incurs some additional work in calculating the length of everything in the cache instead of just throwing it in there.

Reply:
I opted not to use it for now, and just set relatively small number of elements to store in this LRU data structure.